src/planner/expr.go

```diff
@@ -377,6 +375,39 @@ func setAggregatorType(hasAggr, hasDist, isMergeNode bool) aggrType {
 	return nullAgg
 }
 
+//getMatchedTable used to find the real table name for a alias of a subquery
+func getMatchedTable(fieldName string, fields []selectTuple) string {
+	var matchedTb string
+	for _, field := range fields {
+		if (field.alias == "" && field.field == fieldName || field.alias == fieldName) && len(field.referTables) > 0 {
+			if matchedTb == "" {
+				matchedTb = field.referTables[0]
+			} else {
+				return ""
+			}
+		}
+	}
+	return matchedTb
+}
+
+//getMatchedField used to find matched field in the subquery
+func getMatchedField(field selectTuple, fields []selectTuple) (selectTuple, error) {
+	var matchedField selectTuple
+	for _, f := range fields {
+		if (f.alias == "" && field.field == f.field) || (field.field == f.alias) {
+			if matchedField.field == "" {
+				matchedField = f
+			} else {
+				return field, errors.Errorf("unsupported: duplicate.column.name.'%s'", field.field)
```

Comment:
Check duplicate column name can put in `scanAliasedTableExpr`, then if matched  just return.
In `MySQL`:
```
mysql> select b from (select a,a,b from t3)as t;
ERROR 1060 (42S21): Duplicate column name 'a'
```
In `Radon`:
```
mysql> select b from (select a,a,b from t3) as t;
+------+
| b    |
+------+
|    4 |
|    0 |
|    2 |
|    1 |
|    0 |
|    2 |
+------+
6 rows in set (1.11 sec)
```

Reply:
But in mysql  "select a,a,b from t3;"  is legal.

Reply:
`select b from (select a,a,b from t3)as t;` is illegal. 
And every time judge it again, it is repeated Logically.